ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -999,7 +999,7 @@ let on_notification server (notification : Client_notification.t) :
       (* we need [update_text] with no changes to get a new merlin pipeline;
          otherwise the diagnostics don't get updated *)
       let doc = Document.update_text doc [] in
-      Document_store.put store doc;
+      let (_ : Document.t) = Document_store.update store uri ~f:Fun.id in
```

Comment:
hmm, isn't this a bug? we're creating a new doc above but not putting it in the document store

we could merge this line with the above and fix the problem I think

`Document_store.update store uri ~f:(fun d -> Document.update_text d [])`

Reply:
Indeed. Fixes.